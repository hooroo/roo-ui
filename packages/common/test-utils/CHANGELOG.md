# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
