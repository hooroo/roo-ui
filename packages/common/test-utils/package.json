{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
