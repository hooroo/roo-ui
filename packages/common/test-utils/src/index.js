export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
