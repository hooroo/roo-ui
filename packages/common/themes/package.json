{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
