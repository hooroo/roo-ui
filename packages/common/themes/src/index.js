import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
