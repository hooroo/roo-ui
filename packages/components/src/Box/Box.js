import styled from 'styled-components';
import tag from 'clean-tag';
import { space, width, color } from 'styled-system';

const Box = styled(tag)`
  ${space}
  ${width}
  ${color}
`;

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...color.propTypes,
};

export default Box;
