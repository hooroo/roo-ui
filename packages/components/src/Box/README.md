# Box

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Box } from '@roo-ui/components';

export default (
  <Box>Hello world</Box>
);
```

## Customization

This component can be customized with [styled-system](https://jxnblk.com/styled-system) by passing props for
`display`,
`space`,
`width`,
`minWidth`,
`maxWidth`,
`height`,
`minHeight`,
`maxHeight`,
`color`,
`textAlign`,
`boxShadow`,
`borderRadius`,
`borders`,
`borderColor`,
`flexBasis`,
`flex`,
`position` &
`zIndex` [functions](http://jxnblk.com/styled-system/table/)
