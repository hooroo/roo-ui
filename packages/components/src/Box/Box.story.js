import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import Box from './Box';
import README from './README.md';

storiesOf('Components|Box', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Box p={3} bg="grey.6">
      Hello world
    </Box>
  ));
