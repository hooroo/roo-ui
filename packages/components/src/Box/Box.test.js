import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import Box from '.';

describe('<Box />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Box>Hello world</Box>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
