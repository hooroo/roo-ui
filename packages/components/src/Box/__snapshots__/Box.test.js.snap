// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders correctly 1`] = `
<Clean.div
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "width",
      "w",
      "fontSize",
      "f",
      "color",
      "bg",
      "fontFamily",
      "font",
      "textAlign",
      "align",
      "lineHeight",
      "fontWeight",
      "letterSpacing",
      "display",
      "maxWidth",
      "minWidth",
      "height",
      "maxHeight",
      "minHeight",
      "size",
      "ratio",
      "alignItems",
      "alignContent",
      "justifyContent",
      "justify",
      "flexWrap",
      "wrap",
      "flexBasis",
      "flexDirection",
      "flex",
      "justifySelf",
      "alignSelf",
      "order",
      "gridGap",
      "gridColumnGap",
      "gridRowGap",
      "gridColumn",
      "gridRow",
      "gridAutoFlow",
      "gridAutoColumns",
      "gridAutoRows",
      "gridTemplateColumns",
      "gridTemplateRows",
      "border",
      "borderTop",
      "borderRight",
      "borderBottom",
      "borderLeft",
      "borderColor",
      "borderRadius",
      "boxShadow",
      "background",
      "backgroundImage",
      "bgImage",
      "backgroundSize",
      "bgSize",
      "backgroundPosition",
      "bgPosition",
      "backgroundRepeat",
      "bgRepeat",
      "position",
      "zIndex",
      "top",
      "right",
      "bottom",
      "left",
      "hover",
      "focus",
      "active",
      "disabledStyle",
      "textStyle",
      "colors",
      "buttonStyle",
      "borderWidth",
    ]
  }
  className=""
  is="div"
>
  Hello world
</Clean.div>
`;
